--- a/usr/include/CGAL/Straight_skeleton_2/Straight_skeleton_aux.h
+++ b/usr/include/CGAL/Straight_skeleton_2/Straight_skeleton_aux.h
@@ -164,7 +164,7 @@ private:
   Ref_counted_base& operator=( Ref_counted_base const &);
 protected:
   Ref_counted_base(): mCount(0) {}
-  virtual ~Ref_counted_base() {}
+  virtual ~Ref_counted_base() CGAL_NOEXCEPT(CGAL_NO_ASSERTIONS_BOOL) {}
 public:
     void AddRef() const { ++mCount; }
     void Release() const

