diff --git a/usr/include/CGAL/Random_impl.h b/usr/include/CGAL/Random_impl.h
index 89ca7057fe5..d9d83f6b36a 100644
--- a/usr/include/CGAL/Random_impl.h
+++ b/usr/include/CGAL/Random_impl.h
@@ -20,8 +20,8 @@
 #define CGAL_INLINE_FUNCTION
 #endif
 
-#include <ctime>
 #include <iostream>
+#include <random>
 #include <sstream>
 
 namespace CGAL {
@@ -35,10 +35,8 @@ Random::
 Random()
     :  val(0)
 {
-    // get system's time
-    std::time_t s;
-    std::time( &s);
-    seed = (unsigned int)s;
+    std::random_device rd;
+    seed = rd();
     // initialize random numbers generator
     rng.seed(static_cast<boost::int32_t>(seed));
     random_value = get_int(0, 1<<15);
@@ -49,10 +47,8 @@ Random::
 Random(internal::Random_print_seed)
     :  val(0)
 {
-    // get system's time
-    std::time_t s;
-    std::time( &s);
-    seed = (unsigned int)s;
+    std::random_device rd;
+    seed = rd();
     std::cerr << "CGAL::Random()::get_seed() = " << seed << std::endl;
     // initialize random numbers generator
     rng.seed(static_cast<boost::int32_t>(seed));

