--- a/usr/include/CGAL/Tools/chained_map.h
+++ b/usr/include/CGAL/Tools/chained_map.h
@@ -24,10 +24,10 @@ namespace CGAL {
 namespace internal {
 
 template <typename T, typename Allocator = CGAL_ALLOCATOR(T) > class chained_map;
-template <typename T> class chained_map_elem;
+template <typename T> struct chained_map_elem;
 
 template <typename T>
-class chained_map_elem
+struct chained_map_elem
 {
   template<typename T2, typename Alloc> friend class chained_map;
   std::size_t k; T i;

