--- a/usr/include/CGAL/IO/VRML_2_ostream.h
+++ b/usr/include/CGAL/IO/VRML_2_ostream.h
@@ -28,7 +28,12 @@ class VRML_2_ostream {
 public:
     VRML_2_ostream()           : m_os(0)  {}
     VRML_2_ostream(std::ostream& o) : m_os(&o) { header();}
-    ~VRML_2_ostream()  { close(); }
+    ~VRML_2_ostream() CGAL_NOEXCEPT(CGAL_NO_ASSERTIONS_BOOL)
+    {
+      CGAL_destructor_assertion_catch(
+        close();
+      );
+    }
     void open(std::ostream& o)   { m_os = &o; header(); }
     void close() {
         if ( m_os)

