--- a/usr/include/CGAL/Nef_S2/Sphere_map.h
+++ b/usr/include/CGAL/Nef_S2/Sphere_map.h
@@ -229,7 +229,12 @@ public:
   Sphere_map(bool = false) : boundary_item_(boost::none),
     svertices_(), sedges_(), sfaces_(), shalfloop_() {}
 
-  ~Sphere_map() { clear(); }
+  ~Sphere_map() CGAL_NOEXCEPT(CGAL_NO_ASSERTIONS_BOOL)
+  {
+    CGAL_destructor_assertion_catch(
+      clear();
+    );
+  }
 
   Sphere_map(const Self& D) : boundary_item_(boost::none),
     svertices_(D.svertices_),

