diff --git a/usr/include/CGAL/Modifiable_priority_queue.h b/usr/include/CGAL/Modifiable_priority_queue.h
index b6bf5e43cf..2ef0564da8 100644
--- a/usr/include/CGAL/Modifiable_priority_queue.h
+++ b/usr/include/CGAL/Modifiable_priority_queue.h
@@ -102,14 +102,12 @@ public:
 
   boost::optional<value_type> extract_top()
   {
-    boost::optional<value_type> r ;
-    if ( !empty() )
-    {
-      value_type v = top();
-      pop();
-      r = boost::optional<value_type>(v) ;
-    }
-    return r ;
+    if ( empty() )
+      return boost::none;
+
+    value_type v = top();
+    pop();
+    return boost::optional<value_type>(v) ;
   }
 
   static handle null_handle() { return handle(false); }

