--- a/usr/include/CGAL/IO/File_writer_OFF.h
+++ b/usr/include/CGAL/IO/File_writer_OFF.h
@@ -29,8 +29,8 @@ class CGAL_EXPORT File_writer_OFF {
     std::ostream*           m_out;
     File_header_OFF         m_header;
 public:
-    File_writer_OFF( bool verbose = false) : m_header( verbose) {}
-    File_writer_OFF( const File_header_OFF& h) : m_header( h) {}
+    File_writer_OFF( bool verbose = false) : m_out(nullptr), m_header( verbose) {}
+    File_writer_OFF( const File_header_OFF& h) : m_out(nullptr), m_header( h) {}
 
     std::ostream&           out()          { return *m_out;   }
     File_header_OFF&        header()       { return m_header; }

