--- a/usr/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse_impl.h
+++ b/usr/include/CGAL/Surface_mesh_simplification/Detail/Edge_collapse_impl.h
@@ -41,6 +41,8 @@ namespace Surface_mesh_simplification
   ,Get_placement      (aGet_placement)
   ,Visitor            (aVisitor)
   ,m_has_border       (false)
+  ,mInitialEdgeCount  (0)
+  ,mCurrentEdgeCount  (0)
 {
   const FT cMaxDihedralAngleCos = std::cos( 1.0 * CGAL_PI / 180.0 ) ;
 

