--- a/usr/include/CGAL/Nef_S2/SM_overlayer.h
+++ b/usr/include/CGAL/Nef_S2/SM_overlayer.h
@@ -2118,7 +2118,7 @@ complete_face_support(SVertex_iterator v_start, SVertex_iterator v_end,
 { CGAL_NEF_TRACEN("complete_face_support");
   for (SVertex_iterator v = v_start; v != v_end; ++v) {
     CGAL_NEF_TRACEN("VERTEX = "<<PH(v));
-    Mark m_buffer[2];
+    Mark m_buffer[2] {};
     SHalfedge_handle e_below = halfedge_below(v);
     if ( v == v_start ) {
       for (int i=0; i<2; ++i){
