--- a/usr/include/CGAL/Nef_S2/SM_point_locator.h
+++ b/usr/include/CGAL/Nef_S2/SM_point_locator.h
@@ -493,7 +493,7 @@ public:
           Sphere_segment first_half(p,p.antipode(),c);
           Sphere_segment second_part(p.antipode(),p,c);
           if(!do_intersect_internally(ei->circle(), first_half, p_res)) {
-            do_intersect_internally(ei->circle(), second_part, p_res);
+            (void)do_intersect_internally(ei->circle(), second_part, p_res);
           }
         }
 

