--- a/usr/include/CGAL/Nef_3/K3_tree.h
+++ b/usr/include/CGAL/Nef_3/K3_tree.h
@@ -472,7 +472,7 @@ public:
     Segment_3 segment;
     bool initialized;
   public:
-    Objects_around_segment() : initialized(false) {}
+    Objects_around_segment() : root_node(nullptr), initialized(false) {}
     Objects_around_segment( const K3_tree& k, const Segment_3& s) :
       root_node(k.root), segment(s), initialized(true) {
       CGAL_NEF_TRACEN("Objects_around_segment: input segment: "<<segment);
@@ -503,7 +503,8 @@ public:
       CGAL_assertion_code( Segment_3 prev_segment;)
       CGAL_assertion_code( bool first_segment;)
     public:
-      Iterator() : node(0) {}
+      Iterator() : node(nullptr)
+      { CGAL_assertion_code( first_segment = false); }
       Iterator( const Node* root, const Segment_3& s) {
         CGAL_assertion_code( first_segment = true);
         S.push_front( Candidate( root, s));

