--- a/usr/include/CGAL/Handle_for.h
+++ b/usr/include/CGAL/Handle_for.h
@@ -149,12 +149,14 @@ public:
         return *this;
     }
 
-    ~Handle_for()
+    ~Handle_for() CGAL_NOEXCEPT(CGAL_NO_ASSERTIONS_BOOL)
     {
-      if (--(ptr_->count) == 0) {
-        Allocator_traits::destroy(allocator, ptr_);
-        allocator.deallocate( ptr_, 1);
-      }
+      CGAL_destructor_assertion_catch(
+        if (--(ptr_->count) == 0) {
+          Allocator_traits::destroy(allocator, ptr_);
+          allocator.deallocate( ptr_, 1);
+        }
+      );
     }
 
     void

