--- a/usr/include/CGAL/Tools/chained_map.h
+++ b/usr/include/CGAL/Tools/chained_map.h
@@ -24,10 +24,10 @@ namespace CGAL {
 namespace internal {
 
 template <typename T, typename Allocator = CGAL_ALLOCATOR(T) > class chained_map;
-template <typename T> class chained_map_elem;
+template <typename T> struct chained_map_elem;
 
 template <typename T>
-class chained_map_elem
+struct chained_map_elem
 {
   template<typename T2, typename Alloc> friend class chained_map;
   std::size_t k; T i;
@@ -229,7 +229,7 @@ T& chained_map<T, Allocator>::access(Item p, std::size_t x)
 
 template <typename T, typename Allocator>
 chained_map<T, Allocator>::chained_map(std::size_t n, const T& d)
-  : table(nullptr), reserved_size(n), def(d)
+  : table(nullptr), table_end(nullptr), free(nullptr), table_size(0), table_size_1(0), reserved_size(n), def(d)
 {
 }

