--- a/usr/include/CGAL/Nef_2/debug.h
+++ b/usr/include/CGAL/Nef_2/debug.h
@@ -61,19 +61,19 @@
 #define CGAL_NEF_TRACEN(t) if((debugthread%CGAL_NEF_DEBUG)==0) \
  std::cerr<< " "<<t<<std::endl;
 #else
-#define CGAL_NEF_TRACEN(t)
+#define CGAL_NEF_TRACEN(t) (static_cast<void>(0))
 #endif
 
 #ifndef NDEBUG
 #define CGAL_NEF_CTRACE(b,t) if(b) std::cerr<<" "<<t; else std::cerr<<" 0"
 #else
-#define CGAL_NEF_CTRACE(b,t)
+#define CGAL_NEF_CTRACE(b,t) (static_cast<void>(0))
 #endif
 
 #ifndef NDEBUG
 #define CGAL_NEF_CTRACEN(b,t) if(b) std::cerr<<" "<<t<<"\n"; else std::cerr<<" 0\n"
 #else
-#define CGAL_NEF_CTRACEN(b,t)
+#define CGAL_NEF_CTRACEN(b,t) (static_cast<void>(0))
 #endif
 
 #endif // CGAL_NEF_2_DEBUG_H

